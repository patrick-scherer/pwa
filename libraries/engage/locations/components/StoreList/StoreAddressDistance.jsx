// @flow
import React from 'react';
import formatDistance from '../../helpers/formatDistance';
import { storeDistance } from './style';

type Props = {
  distance?: number;
  unitSystem?: string;
}

export const UNIT_SYSTEM_METRIC = 'metric';
export const UNIT_SYSTEM_IMPERIAL = 'imperial';

/**
 * Renders a single store distance.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
const StoreAddressDistance = ({ distance = null, unitSystem = UNIT_SYSTEM_METRIC }: Props) => {
  if (distance === null) {
    return null;
  }

  return (
    <span className={storeDistance}>
      {formatDistance(distance, unitSystem === UNIT_SYSTEM_IMPERIAL)}
    </span>
  );
};

StoreAddressDistance.defaultProps = {
  distance: null,
  unitSystem: UNIT_SYSTEM_METRIC,
};

export default StoreAddressDistance;
