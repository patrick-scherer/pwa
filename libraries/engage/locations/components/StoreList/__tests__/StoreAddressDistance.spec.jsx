import React from 'react';
import { shallow } from 'enzyme';
import formatDistance from '../../../helpers/formatDistance';
import StoreAddressDistance, { UNIT_SYSTEM_METRIC, UNIT_SYSTEM_IMPERIAL } from '../StoreAddressDistance';

jest.mock('../../../helpers/formatDistance', () => jest.fn());

describe('<StoreAddressDistance', () => {
  beforeAll(() => {
    formatDistance.mockImplementation((distance, imperial) => `${distance} ${imperial ? UNIT_SYSTEM_IMPERIAL : UNIT_SYSTEM_METRIC}`);
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should not render when no distance was passed', () => {
    const wrapper = shallow((<StoreAddressDistance />));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.isEmptyRender()).toEqual(true);
  });

  it('should render when the distance is 0', () => {
    const distance = 0;
    const unitSystem = UNIT_SYSTEM_METRIC;
    const wrapper = shallow((<StoreAddressDistance distance={distance} />));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('span').text()).toBe(`${distance} ${unitSystem}`);
    expect(formatDistance).toHaveBeenCalledTimes(1);
    expect(formatDistance).toHaveBeenCalledWith(distance, false);
  });

  it('should render when a unit system is set', () => {
    const distance = 5.4;
    const unitSystem = UNIT_SYSTEM_IMPERIAL;
    const wrapper = shallow((
      <StoreAddressDistance
        distance={distance}
        unitSystem={unitSystem}
      />
    ));
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('span').text()).toBe(`${distance} ${unitSystem}`);
    expect(formatDistance).toHaveBeenCalledTimes(1);
    expect(formatDistance).toHaveBeenCalledWith(distance, true);
  });
});
