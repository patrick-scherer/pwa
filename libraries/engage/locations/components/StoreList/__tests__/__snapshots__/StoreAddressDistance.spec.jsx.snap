// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreAddressDistance should not render when no distance was passed 1`] = `""`;

exports[`<StoreAddressDistance should render when a unit system is set 1`] = `
<span
  className="css-1fxtar0"
>
  5.4 imperial
</span>
`;

exports[`<StoreAddressDistance should render when the distance is 0 1`] = `
<span
  className="css-1fxtar0"
>
  0 metric
</span>
`;
